todo_app/app.py

```diff
@@ -19,72 +18,50 @@ def create_app(debug=True):
 
     @app.route('/', methods = ["GET"])
     def index():
-        context = 'app.py index route'
-        doing = 'trello_cards'
-        #items = get_items()
-        #items = get_trello_cards()
-        trello_items = get_trello_cards()
-        writelog(context, doing, 'trello_items',trello_items)
+        mongo_items = get_mongo_items()
         items = []
-        doing = 'building items'
-        writelog(context, doing, 'items',items)
-        for item in trello_items:
-            #writelog(context, doing, 'item',item)
-            #items.append(Item(id=item_id, name=item_name, status=item_list))
-            items.append(Item(item['id'], item['name'], item['idList']))
-            writelog(context, doing, 'items',items)
+        for item in mongo_items:
+            items.append(Item(item['_id'], item['name'], item['idlist']))
         item_view_model = ViewModel(items)
-        #item_view_model = items
         return render_template('index.html', view_model=item_view_model)
-        # return render_template('index.html', items_list = items)
 
     @app.route('/add-item', methods = ["POST"])
     def add_todo_item():
         item_title = request.form.get('title')
-        #add_item(item_title)
-        add_trello_card(item_title)
-        context = 'app.py add-item route'
-        doing = 'add_trello_card'
-        writelog(context, doing, 'item_title',item_title)
+        add_mongo_item(item_title)
+        #context = 'app.py add-item route'
+        #doing = 'add_trello_card'
+        #writelog(context, doing, 'item_title',item_title)
         return redirect('/')
 
     @app.route('/del-item', methods = ["POST"])
     def del_todo_item():
         exis_item_title = request.form.get("selected_item")
-        #id = get_item_id_by_title(exis_item_title)
-        id = get_trello_card_id_by_name(exis_item_title)
-        #delete_item(id)
-        del_trello_card(id)
+        id = get_mongo_item_id_by_name(exis_item_title)
         context = 'app.py del-item route'
-        doing = 'del_trello_card'
+        doing = 'del_mongo_card'
+        writelog(context, doing, 'exis_item_title',exis_item_title)
+        del_mongo_item(id)
+        context = 'app.py del-item route'
+        doing = 'del_mongo_card'
         writelog(context, doing, 'id',id)
         return redirect('/')
 
     @app.route('/upd-item', methods = ["POST"])
     def upd_todo_item():
         exis_item_title = request.form.get("selected_item")
-        #exis_item = get_item_by_title(exis_item_title)
-        exis_item = get_trello_card_by_name(exis_item_title) 
-        #exis_item_status = exis_item['status']
-        exis_item_status = exis_item['idList']
+        exis_item = get_mongo_item_by_name(exis_item_title) 
```

Comment:
I know this builds on your Trello approach, but I'd always advise to be wary of "get by name" type approaches - they typically ask for problems like "if we have two cards with the same name, what happens?". Working by Id here should be less bug-prone

Reply:
I'll put a comment below on the HTML for how we might do that